### ISSUE TITLE
SC2154: Option to ignore the capitals special-case

### ISSUE THREAD

**Issue by viraptor** [koalaman/shellcheck#930]
#### For bugs
- Rule Id: SC2154
- My shellcheck version (`shellcheck --version` or "online"): 0.4.6 and online
- [ ] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
echo "#{SOME_VAR}"
```

#### Here's what shellcheck currently says:
```
No issues detected!
```

#### Here's what I wanted or expected to see:
I expected an option which stops treating the capitalised variables as special. I'd like to enable undefined variables reports on both lowercase and uppercase variables.
